pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,260 @@
+"""
+Signal processing tools for Pynapple.
+
+Contains functionality for signal processing pynapple object; fourier transforms and wavelet decomposition.
+"""
+
+from itertools import repeat
+from math import ceil, floor
+
+import numpy as np
+from scipy.signal import welch
+
+import pynapple as nap
+
+
+def compute_spectrum(sig, fs=None):
+    """
+    Performs numpy fft on sig, returns output
+    ..todo: Make sig handle TsdFrame, TsdTensor
+
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float, optional
+        Sampling rate, in Hz. If None, will be calculated from the given signal
+    """
+    if not isinstance(sig, nap.Tsd):
+        raise TypeError("Currently compute_fft is only implemented for Tsd")
+    if fs is None:
+        fs = sig.index.shape[0] / (sig.index.max() - sig.index.min())
+    fft_result = np.fft.fft(sig.values)
+    fft_freq = np.fft.fftfreq(len(sig.values), 1 / fs)
+    return fft_result, fft_freq
+
+
+def compute_welch_spectrum(sig, fs=None):
+    """
+    Performs scipy Welch's decomposition on sig, returns output
+    ..todo: Make sig handle TsdFrame, TsdTensor
+
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float, optional
+        Sampling rate, in Hz. If None, will be calculated from the given signal
+    """
+    if not isinstance(sig, nap.Tsd):
+        raise TypeError("Currently compute_fft is only implemented for Tsd")
+    if fs is None:
+        fs = sig.index.shape[0] / (sig.index.max() - sig.index.min())
+    freqs, spectogram = welch(sig.values, fs=fs)
```

Comment:
In the notebook I would do a comparison of the difference between calling welch fft versus fft